### ISSUE TITLE
Tuple not being destructured in AST

### ISSUE THREAD

**Issue by alfonsogarciacaro** [fsharp/FSharp.Compiler.Service#543]
This issue has been reported in Fable: https://github.com/fsprojects/Fable/issues/55

If you have the following code:
```fsharp
let start name =
    name, "1"

let last (name, values) =
    printf "Last: Name='%s', Values='%s'" name values

let test() =
    start "Test" |> last
```

The generated AST for `test` method's body is:
```
Call
  (null,val op_PipeRight,[],
   [type Microsoft.FSharp.Core.string * Microsoft.FSharp.Core.string;
    type Microsoft.FSharp.Core.unit],
   [Call
  (null,val start,[],[type Microsoft.FSharp.Core.string],
   [Const ("Test",type Microsoft.FSharp.Core.string)]);
    Value val last])
```

The problem is in the last line, the result of `start` is applied as is to `last` when it should be destructured in advance. Interestingly, if the string format in `last` changes to `"Last: Name='%A', Values='%s'"`, the correct AST is generated. May this indicate a problem with the type checker?

```
Call
  (null,val op_PipeRight,[],
   [type Microsoft.FSharp.Core.string * Microsoft.FSharp.Core.string;
    type Microsoft.FSharp.Core.unit],
   [Call
  (null,val start,[],[type Microsoft.FSharp.Core.string],
   [Const ("Test",type Microsoft.FSharp.Core.string)]);
    Lambda
  (val tupledArg,
   Let
  ((val name,
    TupleGet
  (type Microsoft.FSharp.Core.string * Microsoft.FSharp.Core.string,0,
   Value val tupledArg)),
   Let
  ((val values,
    TupleGet
  (type Microsoft.FSharp.Core.string * Microsoft.FSharp.Core.string,1,
   Value val tupledArg)),
   Call
  (null,val last,[],[type Microsoft.FSharp.Core.string],
   [Value val name; Value val values]))))])
```

Another way to fix the issue is removing the piping in `test`, like `last(start "Test")`, which may somehow relate this to the second problem described in #531.

Any hint on how to fix this issue would be much appreciated.

**Comment by dsyme** [comment]
This is a serious bug in the expression tree support, it definitely needs to be fixed.

**Comment by dsyme** [comment]
Fixed by https://github.com/fsharp/FSharp.Compiler.Service/pull/549

**Comment by alfonsogarciacaro** [comment]
Fantastic! Thanks a lot for that, Don. Looking forward to the release!

**Comment by dsyme** [comment]
Fix is in 2.0.0.7
